server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTarget.java

```diff
@@ -41,45 +62,49 @@ public SegmentReplicationTarget(
         super("replication_target", indexShard, new ReplicationLuceneIndex(), listener);
         this.checkpoint = checkpoint;
         this.source = source;
-        this.state = new SegmentReplicationState();
+        this.state = new SegmentReplicationState(stateIndex);
+        this.multiFileWriter = new MultiFileWriter(indexShard.store(), stateIndex, getPrefix(), logger, this::ensureRefCount);
     }
 
     @Override
     protected void closeInternal() {
-        // TODO
+        try {
+            multiFileWriter.close();
+        } finally {
+            // free store. increment happens in constructor
```

Comment:
I don't see it in the ctor, is it in the parent class?  I think we should consider having subclasses manage these refCounts or the parent, but mixing is confusing.